pennylane/devices/default_qubit.py

```diff
@@ -67,7 +67,7 @@ def stopping_condition(op: qml.operation.Operator) -> bool:
     if op.__class__.__name__[:3] == "Pow" and qml.operation.is_trainable(op):
         return False
 
-    return op.has_matrix
+    return isinstance(op, (Conditional, MidMeasureMP)) or op.has_matrix
```

Comment:
What happens if we have `Conditional(m0, qml.Permute([2,1,0], wires=(0,1,2) )`?

Reply:
I would think that the operation will remain in the tape and if `m0 == 1` then `qml.Permute([2,1,0], wires=(0,1,2) )` will be applied to the state vector. Am I missing something?

Reply:
I think Christina is correct. This function checks whether or not an operator is supported by `default.qubit`. With Christina's example, the operator will be considered supported since it's a `Conditional`, but `apply_operation` will fail since `Permute` is unsupported. So we should have a separate condition for `Conditional` that also verifies that the base operator is supported.

Reply:
> Permute is unsupported

Ah, that's what I was missing. Make sense now :)